src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animals) {
+    if (animals.hidden === false) {
+      animals.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter(animal => animal.health <= 0);
```

Comment:
You don't need to filter every time when you `bite` method was called, let's check health of animal and then filter